scripts/check_golang_version.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+GOLANG_VERSION=$(./scripts/get_golang_version.sh)
+
+# Check to make sure that it matches what is specified in go.mod
+
+GOMOD_VERSION=$(go mod edit -json | jq -r ".Go")
+TRIMMED_GOLANG_VERSION=$(awk -F. '{ print $1 "." $2 }' <<< "${GOLANG_VERSION}")
+
+if ! [[ "${GOLANG_VERSION}" == "${GOMOD_VERSION}" || "${TRIMMED_GOLANG_VERSION}" == "${GOMOD_VERSION}" ]]
```

Comment:
Do we want == or >=
Is 1.14.8 allowed? IS 1.15.x allowed?

Reply:
My understanding was that thought was to pin this to a precise version.

( Note that most of this file is existing code. )

Reply:
For automated systems maybe but for human developers we should certainly set a lower bound and allow newer things to be okay

Reply:
Ok, that's a good point.

Reply:
The pinning should be for our build system.. not for other cases.
I think that it's legit if we fail to build with a newer version due to breaking changes.

Reply:
I think we would prefer to pin to the major version specified in go.mod (1.14 currently), no? I don't think this should be hardcoded to 1.12.

Reply:
> I think we would prefer to pin to the major version specified in go.mod (1.14 currently), no? I don't think this should be hardcoded to 1.12.

The hardcoded version (1.12) is the first version of go that fully supports the go modules command that we use to get the pinned version from `go.mod`, so we need to check this first to disambiguate that failure from the comparison check and possible error later in the script.